types/params.go

```diff
@@ -142,9 +142,13 @@ type SynchronyParams struct {
 // parameters are set or become insufficient to preserve liveness. Refer to
 // https://github.com/cometbft/cometbft/issues/2184 for more details.
 func (sp SynchronyParams) InRound(round int32) SynchronyParams {
+	messageDelay := time.Duration(math.Pow(1.1, float64(round)) * float64(sp.MessageDelay))
+	if messageDelay < 0 { // overflow check: needed for some architectures
```

Comment:
Can we guarantee that we don't overflow into a positive range here?

Checking overflow with MaxInt64 seems safer.

```go
func (sp SynchronyParams) InRound(round int32) SynchronyParams {
	scalingFactor := math.Pow(1.1, float64(round))
	var messageDelay time.Duration
	if scalingFactor*float64(sp.MessageDelay) <= float64(math.MaxInt64) {
		messageDelay = time.Duration(scalingFactor * float64(sp.MessageDelay))
	} else {
		messageDelay = time.Duration(math.MaxInt64)
	}
	return SynchronyParams{
		Precision:    sp.Precision,
		MessageDelay: messageDelay,
	}
}
```

Reply:
rounds are always inc by 1. but I guess if someone sets message delay to some really big number than it's possible.

Reply:
The undefined behavior is coming from casting a too large float64 to a int64 which is undefined in Go. The goal here is basically to avoid the overflow entirely, and make explicit the behavior we want.

Reply:
Here is an example of the issue we are running into. It doesn't have anything to do with time.Duration. It's just an issue of converting between float64 and int64 when `>= float64(math.MaxInt64)`.

Go Example: https://ideone.com/x7VPF1
C Example: https://ideone.com/0aQJhZ

Reply:
I will repeat my previous comment here:

So, this is a workaround, not a solution.

The real solution is to add some "safe add duration" method, say, to `types/time` package (typically imported as `cmttime`?